### PR TITLE
Add the ability to disable logging to file with a command line option

### LINKED ISSUE #3660: Add option to disable logging to file
It would be convenient to have an option to disable logging to files.

### PR DESCRIPTION
**Pull request by worktycho**
Fixes #3660

### CODE REVIEW

File: src/Root.cpp [hunk 1]
```diff
@@ -116,13 +116,17 @@ void cRoot::Start(std::unique_ptr<cSettingsRepositoryInterface> a_OverridesRepo)
...
 	{
-		LOGERROR("Failed to open log file, aborting");
-		return;
+		auto fileLogListenerRet = MakeFileListener();
+		if (!fileLogListenerRet.first)
+		{
+			LOGERROR("Failed to open log file, aborting");
+			return;
+		}
+		auto fileAttachment = cLogger::GetInstance().AttachListener(std::move(fileLogListenerRet.second));
```
Comment (madmaxoft) [inline src/Root.cpp]: Won't this remove the attachment upon exiting the block?

File: src/Logger.h [hunk 2]
```diff
@@ -27,14 +27,15 @@ class cLogger
...
+		cAttachment() : m_listener(nullptr) {}
 		cAttachment(cAttachment && a_other)
 			: m_listener(a_other.m_listener)
 		{
 		}
 
 		~cAttachment()
 		{
-			cLogger::GetInstance().DetachListener(m_listener);
+			if (m_listener != nullptr) cLogger::GetInstance().DetachListener(m_listener);
```
Comment (madmaxoft) [inline src/Logger.h]: Please do use a full curly-brace block for controlled statements.

**Comment by mathiascode** [comment]
What's the status of this PR?

**Comment by Seadragon91** [comment]
Will fix conflict and merge this.

File: src/Root.cpp [hunk 3]
```diff
@@ -116,14 +116,19 @@ void cRoot::Start(std::unique_ptr<cSettingsRepositoryInterface> a_OverridesRepo)
...
+	cLogger::cAttachment fileAttachment;
+	if (!a_OverridesRepo->HasValue("Server","DisableLogFile"))
 	{
-		m_TerminateEventRaised = true;
-		LOGERROR("Failed to open log file, aborting");
-		return;
+		auto fileLogListenerRet = MakeFileListener();
+		if (!fileLogListenerRet.first)
+		{
+			m_TerminateEventRaised = true;
```
Comment (Seadragon91) [inline src/Root.cpp]: This line was in master and caused the conflict, re-added it here.

**Pull request merged**